### PR TITLE
Use a template instead of regex to set hosts file

### PR DESCRIPTION
**Pull request by TeaBough**
Should close issue #689 and #680

### CODE REVIEW

File: roles/common/templates/hosts.j2 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
+::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
+{% for item in groups['all'] %}
+{% if hostvars[item].private_ipv4 is defined %}
+{{ hostvars[item].private_ipv4 }} {{ item }}{% if use_host_domain %} {{ item }}.{{ host_domain }}{% endif %}
```
Comment (eirslett) [inline roles/common/templates/hosts.j2]: We really need #680 in here as well!

Comment (TeaBough) [inline roles/common/templates/hosts.j2]: Done

**Comment by eirslett** [comment]
+1

**Comment by siddharthist** [comment]
I'll test this today.

**Comment by siddharthist** [comment]
I'm getting an error that I think is unrelated:
```
TASK: [glusterfs | enable community repo] ************************************* 
failed: [lb-mi-control-02] => {"failed": true}
msg: Failed to validate the SSL certificate for download.gluster.org:443. Use validate_certs=no or make sure your managed systems have a valid CA certificate installed. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible
failed: [lb-mi-control-01] => {"failed": true}
msg: Failed to validate the SSL certificate for download.gluster.org:443. Use validate_certs=no or make sure your managed systems have a valid CA certificate installed. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible
failed: [lb-mi-control-03] => {"failed": true}
msg: Failed to validate the SSL certificate for download.gluster.org:443. Use validate_certs=no or make sure your managed systems have a valid CA certificate installed. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible
```

**Comment by siddharthist** [comment]
With this PR:
```
[centos@lb-mi-control-01 ~]$ cat /etc/hosts
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
10.0.0.2  lb-mi-worker-002

10.0.0.5  lb-mi-worker-001

10.0.0.6  lb-mi-control-02

10.0.0.4  lb-mi-control-01

10.0.0.3  lb-mi-control-03
```

**Comment by siddharthist** [comment]
Reproducable on both worker and control nodes on AWS as well.

**Comment by siddharthist** [comment]
Works on Vagrant as well, with `default` being the local hostname.

**Comment by TeaBough** [comment]
I've made a little change by removing the newline between hosts,

**Comment by loshkovskyi** [comment]
Tested on OpenStack - looks good.

**Comment by loshkovskyi** [comment]
I faced with a problem that the hosts entries are duplicated when I rerun the playbooks.

**Comment by TeaBough** [comment]
This is odd since it's a template.It basically just loop on all machines that are in the group `all`.

**Comment by loshkovskyi** [comment]
Sorry, I deployed from a wrong branch. It works correctly, with no duplicated entries.

**Comment by ryane** [comment]
lgtm, fixes #689

**Pull request merged**